### PR TITLE
Merge develop to master for release 0.5.1

### PR DESCRIPTION
**Pull request by amandarichardsonn**
Merge develop into master

### CODE REVIEW

File: doc/developer/testing.rst [hunk 1]
```diff
@@ -9,6 +9,88 @@ Quick instructions
...
+  tests/python/test_nonkeyed_cmd.py::test_config_get_command_DNE PASSED                                          [ 89%]
+  tests/python/test_nonkeyed_cmd.py::test_save_command PASSED                                                    [ 89%]
+
+2. Launch a backend database.
+
+Here, we'll set up a small Redis cluster with three nodes on the local
+host at ports `6390`, `6391`, and `6392`, using the ``launch_redis.py`` script located in
+the ``utils`` folder.
+
+.. note:: If the ``nodes`` argument is not one, the launched nodes will be
```
Comment (amandarichardsonn) [inline doc/developer/testing.rst:41]: .. note::
    tab the note or it will not render correctly

**Review by al-rigazzi — APPROVED** [review]
LGTM!

File: doc/developer/testing.rst [hunk 2]
```diff
@@ -9,6 +9,88 @@ Quick instructions
...
+When running the full test suite through the automated SmartRedis test system, several
+steps, are performed on your behalf that must be done manually in order to run an
+individual test case. These steps are as follows:
+
+1. Identify the test you wish to run.
+
+This is most easily done by running the full test suite
+and noting the name and path to the test. For example to run the ``test_config_set_command_DNE``
+from within ``test_nonkeyed_cmd.py``, we'll save off the path:
+``tests/python/test_nonkeyed_cmd.py::test_config_set_command_DNE``
```
Comment (ashao) [inline doc/developer/testing.rst:25]: Put this in a code block

File: doc/changelog.rst [hunk 3]
```diff
@@ -1,10 +1,64 @@
...
-------------------
+0.5.1
+-----
+
+Released on February 15, 2024
+
+Description
+
+- Fix bug when sending an array view
+- Add concurrency groups
```
Comment (ashao) [inline doc/changelog.rst:12]: Use concurrency groups during Github Actions testing

File: doc/install/python_client.rst [hunk 4]
```diff
@@ -27,8 +27,7 @@ the ``Client`` from ``smartredis`` as follows:
 
 .. code-block:: python
```
Comment (amandarichardsonn) [inline doc/install/python_client.rst:28]: change .. code-block:: bash

**Review by ashao — CHANGES_REQUESTED** [review]
Documentation fixes

File: doc/developer/testing.rst (hunk 2 above)
Comment (ashao) [reply doc/developer/testing.rst:25]: Actually this is ok disegard

**Review by ashao — APPROVED** [review]

**Pull request merged**